arrivals-api/src/main/scala/com/pagerduty/arrivals/api/proxy/HttpProxy.scala

```diff
@@ -1,10 +1,16 @@
 package com.pagerduty.arrivals.api.proxy
 
 import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
+import com.pagerduty.akka.http.support.RequestMetadata
+
 import scala.concurrent.Future
 
 // HttpProxy drives a given HttpRequest through an Upstream's Request Lifetime.
 // see ./Upstream.scala for the steps an HttpProxy needs to take to process an upstream request.
 trait HttpProxy[AddressingConfig] {
-  def apply(request: HttpRequest, upstream: Upstream[AddressingConfig]): Future[HttpResponse]
+  def apply(
+      request: HttpRequest,
+      upstream: Upstream[AddressingConfig]
+    )(implicit reqMeta: RequestMetadata
```

Comment:
can the implementer who needs a RequestMetadata just create one with `RequestMetadata.fromRequest(request)` since we are passing in the `HttpRequest`, this way we don't need to pipe this metadata around.

Reply:
Done, but we should fix the rest of the lib in a different PR.